src/rust/engine/graph/src/lib.rs

```diff
@@ -768,36 +865,62 @@ mod tests {
     }
   }
 
+  ///
+  /// A context that keeps a count of spawned Nodes. This is a very basic way to confirm that Nodes
+  /// have (re-)run.
+  ///
+  /// TODO: This is potentially error prone, but is much simpler than actually tracking precisely
+  /// which Nodes ran and in which order. Future work.
+  ///
   #[derive(Clone)]
   struct TContext {
     graph: Arc<Graph<TNode>>,
+    spawn_count: Arc<Mutex<usize>>,
     entry_id: Option<EntryId>,
   }
   impl NodeContext for TContext {
     type Node = TNode;
     fn clone_for(&self, entry_id: EntryId) -> TContext {
       TContext {
         graph: self.graph.clone(),
+        spawn_count: self.spawn_count.clone(),
         entry_id: Some(entry_id),
       }
     }
 
     fn graph(&self) -> &Graph<TNode> {
       &self.graph
     }
+
+    fn spawn<F>(&self, future: F)
+    where
+      F: Future<Item = (), Error = ()> + Send + 'static,
+    {
+      // Avoids introducing a dependency on a threadpool.
```

Comment:
Why is that something we want to avoid doing?

Reply:
No great reason other than complexity... there is an argument to be made in general for doing the simplest thing that works, and this does. But I've since introduced a `[dev-dependency]` for the tests in #6059 and that was relatively painless, so who knows.

Reply:
Got it, was just wondering if there was something deeper. This makes sense.